src/auto_pickup.cpp

```diff
@@ -771,7 +771,11 @@ void auto_pickup::load(const bool bCharacter)
 
     if( !fin.good() ) {
         if (load_legacy(bCharacter)) {
-            save(bCharacter);
+            if (save(bCharacter)) {
+                if(file_exist(sFile)) {
```

Comment:
Checking the existence is not necessary, `remove_file` works fine even when the file does not exist. It will simply return `false` (as deleting the file failed), but you're not checking the return value anyway, so it won't make any difference.